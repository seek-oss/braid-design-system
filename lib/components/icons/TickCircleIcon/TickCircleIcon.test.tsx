import React from 'react';
import { TickCircleIcon } from '../..';
import { render } from 'react-testing-library';
import forEachTheme from '../../../test/utils/forEachTheme';

describe('TickCircleIcon', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('Default', () => {
      const { container } = render(
        <ThemeProvider>
          <TickCircleIcon />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Sized', () => {
      const { container } = render(
        <ThemeProvider>
          <TickCircleIcon size="large" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Inline', () => {
      const { container } = render(
        <ThemeProvider>
          <TickCircleIcon inline />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
