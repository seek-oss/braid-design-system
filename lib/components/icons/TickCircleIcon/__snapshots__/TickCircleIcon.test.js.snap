// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickCircleIcon Theme: jobStreet Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: jobStreet Inline 1`] = `
<svg
  class="width__standardTextInline height__standardTextInline Icon__inline Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: jobStreet Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: seekAnz Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: seekAnz Inline 1`] = `
<svg
  class="width__standardTextInline height__standardTextInline Icon__inline Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: seekAnz Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: seekAsia Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: seekAsia Inline 1`] = `
<svg
  class="width__standardTextInline height__standardTextInline Icon__inline Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: seekAsia Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: wireframe Default 1`] = `
<svg
  class="width__standardText height__standardText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: wireframe Inline 1`] = `
<svg
  class="width__standardTextInline height__standardTextInline Icon__inline Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;

exports[`TickCircleIcon Theme: wireframe Sized 1`] = `
<svg
  class="width__largeText height__largeText Box__root borderColor__standard"
  height="16"
  viewBox="0 0 16 16"
  width="16"
>
  <path
    d="M8 .5C3.86.5.5 3.858.5 8c0 4.142 3.358 7.5 7.5 7.5 4.142 0 7.5-3.358 7.5-7.5C15.5 3.858 12.142.5 8 .5zm4.47 5.328l-5.59 5.98c-.14.152-.34.238-.546.238-.208 0-.405-.086-.547-.237L3.537 9.41c-.284-.303-.268-.777.034-1.06.303-.283.777-.27 1.062.033l1.7 1.814 5.042-5.397c.282-.303.758-.318 1.06-.036.302.284.32.758.036 1.06z"
  />
</svg>
`;
